### PR TITLE
[SPARK-19249] Update download page with archived release location

### PR DESCRIPTION
**Pull request by lresende**

### CODE REVIEW

**Comment by lresende** [comment]
Note that I am including only related modified changes, but actually git status list a bunch of other files with mostly stylistics changes on the jekyll generated code like :

-      <li>announcing the release </li>
+      <li>announcing the release</li>

File: downloads.md [hunk 1]
```diff
@@ -68,6 +68,10 @@ Once you've downloaded Spark, you can find instructions for installing and build
 
 <ul id="sparkReleaseNotes"></ul>
 
+### Archived Releases
+
+As new Spark releases comes out for each development stream, previous ones will be archived but they are still available at [Spark release archives](http://archive.apache.org/dist/spark/).
```
Comment (srowen) [inline downloads.md]: releases comes -> releases come
archived but -> archived, but
Also for good measure, make the link an https URL? It works and I suppose we should point people to a secure URL for obtaining software.

**Comment by lresende** [comment]
Comments addressed

**Comment by srowen** [comment]
@lresende there are no further changes. Did you not push?

**Comment by lresende** [comment]
@srowen I had pushed with --force, that's why you don't see additional commits

**Comment by srowen** [comment]
Yes I see it now. You won't need to `--force` unless there's a `rebase` but it is not a problem either way.

**Pull request merged**